packages/apps/plugins/plugin-space/src/components/SpaceMain/SpaceMembersSection.tsx

```diff
@@ -0,0 +1,200 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { CaretDown, Check, UserPlus, UsersThree } from '@phosphor-icons/react';
+import React, { useCallback, useState } from 'react';
+
+import { LayoutAction, useIntent } from '@dxos/app-framework';
+import { type Space, useMembers, SpaceMember, useSpaceInvitations } from '@dxos/react-client/echo';
+import { type CancellableInvitationObservable, InvitationEncoder } from '@dxos/react-client/invitations';
+import { Invitation } from '@dxos/react-client/invitations';
+import { Button, ButtonGroup, DropdownMenu, List, useTranslation } from '@dxos/react-ui';
+import { descriptionText, getSize, mx } from '@dxos/react-ui-theme';
+import { InvitationListItem, IdentityListItem } from '@dxos/shell/react';
+
+import { SPACE_PLUGIN } from '../../meta';
+
+// TODO(thure): Sync with shell?
+const activeActionKeyStorageKey = 'dxos:react-shell/space-manager/active-action';
+
+const Presence = SpaceMember.PresenceState;
+
+const handleCreateInvitationUrl = (invitationCode: string) => `${origin}?spaceInvitationCode=${invitationCode}`;
+
+const SpaceMemberList = ({ members }: { members: SpaceMember[] }) => {
+  return members.length > 0 ? (
+    <List classNames='col-start-2 col-end-5 gap-y-1 grid grid-cols-subgrid items-center'>
+      {members.map((member) => (
+        <IdentityListItem
+          classNames='contents'
+          key={member.identity.identityKey.toHex()}
+          identity={member.identity}
+          presence={member.presence}
+        />
+      ))}
+    </List>
+  ) : null;
+};
+
+export const SpaceMembersSection = ({ space }: { space: Space }) => {
+  const { t } = useTranslation(SPACE_PLUGIN);
+  const invitations = useSpaceInvitations(space.key);
+  const { dispatch } = useIntent();
+
+  const handleCloseDialog = () =>
+    dispatch({ action: LayoutAction.SET_LAYOUT, data: { element: 'dialog', state: false } });
+
+  const handleInvitationSelect = ({
```

Comment:
I think we should move forward with this for now, but something to consider is, while there may be different types of member lists with various information, perhaps shell owns invitations? Instead of managing invitations on this page there could just be an invite/share button which opens the shell.

Reply:
There’s some real utility in not depending on dialogs for these tasks — I’m hoping this experiment will show that if we want tidier code, we may need shell to be able to own these operations in a way that doesn’t also restrict its UI to a dialog.